plugins/logs/plugin.go

```diff
@@ -440,8 +453,10 @@ func (p *Plugin) Start(ctx context.Context) error {
 func (p *Plugin) Stop(ctx context.Context) {
 	p.logger.Info("Stopping decision logger.")
 
-	if _, ok := ctx.Deadline(); ok && p.config.Service != "" {
-		p.flushDecisions(ctx)
+	if *p.config.Reporting.Trigger == plugins.TriggerPeriodic {
+		if _, ok := ctx.Deadline(); ok && p.config.Service != "" {
+			p.flushDecisions(ctx)
+		}
```

Comment:
I can see how that code was here before, I'm trying to understand it: `ctx.Deadline()` returns `(time.Time, bool)`:
```
	// Deadline returns the time when work done on behalf of this context
	// should be canceled. Deadline returns ok==false when no deadline is
	// set. Successive calls to Deadline return the same results.
```
So _if there is a deadline_, that may have been reached or not, we flushDecisions? Why would we not do that for a context that has no deadline...?  🤔